omod/src/main/java/org/openmrs/module/pihcore/fragment/controller/dashboardwidgets/PregnancyIndicatorsFragmentController.java

```diff
@@ -0,0 +1,157 @@
+            obsWithinProgram = PihCoreUtils.getObsWithinProgram(patient, new HashSet<>(Arrays.asList(question)), null, null);
+        }
+        fields.put("deliveryType", SimpleObject.create(
+                "label", "pihcore.latest.delivery.type",
+                "obs", (obsWithinProgram!=null && obsWithinProgram.size() > 0 ) ? obsWithinProgram.get(0) : ""
+        ));
+
+        EncounterSearchCriteriaBuilder searchCriteriaBuilder = new EncounterSearchCriteriaBuilder();
+        searchCriteriaBuilder.setPatient(patient);
+        searchCriteriaBuilder.setEncounterTypes(new HashSet<>(Arrays.asList(emrApiProperties.getAdmissionEncounterType())));
+        List<Encounter> encounters = encounterService.getEncounters(searchCriteriaBuilder.createEncounterSearchCriteria());
+        Date lastAdmissionDatetime = null;
+        for (Encounter encounter : encounters) {
+            Date encounterDatetime = encounter.getEncounterDatetime();
+            if ( lastAdmissionDatetime == null || lastAdmissionDatetime.before(encounterDatetime)) {
+                // find the most recent Admission datetime
+                lastAdmissionDatetime = encounterDatetime;
+            }
+        }
+        fields.put("latestAdmission", SimpleObject.create(
+                "label", "pihcore.latest.admission",
+                "obs",  lastAdmissionDatetime != null ? lastAdmissionDatetime : ""
+        ));
+
+        Set<EncounterType> ancEncTypes = new HashSet<>();
+        ancEncTypes.add(encounterService.getEncounterTypeByUuid(PihEmrConfigConstants.ENCOUNTERTYPE_ANC_INTAKE_UUID));
+        ancEncTypes.add(encounterService.getEncounterTypeByUuid(PihEmrConfigConstants.ENCOUNTERTYPE_ANC_FOLLOWUP_UUID));
+
+        searchCriteriaBuilder = new EncounterSearchCriteriaBuilder();
+        searchCriteriaBuilder.setPatient(patient);
+        searchCriteriaBuilder.setEncounterTypes(ancEncTypes);
+        if ( programUuid != null ) {
+            searchCriteriaBuilder.setFromDate(PihCoreUtils.getProgramEnrollmentDate( patient, programUuid ));
+        }
+        encounters = encounterService.getEncounters(searchCriteriaBuilder.createEncounterSearchCriteria());
+        int numberOfVisits = encounters != null ? encounters.size() : 0;
+        for (Encounter encounter : encounters) {
+            if (encounter.getEncounterType().getUuid().equals(PihEmrConfigConstants.ENCOUNTERTYPE_ANC_INTAKE_UUID)) {
+                for (Obs candidate : encounter.getObsAtTopLevel(false)) {
+                    if (candidate.getConcept().getUuid().equalsIgnoreCase(SierraLeoneConfigConstants.CONCEPT_NUMBEROFANCVISITS_UUID)) {
+                        Double valueNumeric = candidate.getValueNumeric();
+                        if (!valueNumeric.isNaN()) {
+                            // the question on the form is "What number ANC Visit is this?", therefore it should include the ANC Intake visit
+                            numberOfVisits = numberOfVisits + valueNumeric.intValue() - 1;
```

Comment:
I understand the logic here but will be interesting to see how this works out in practice.  The idea is that ideally this will always be "1" (the Intake is the first visit?) but this is to capture the case where somehow there were visits prior to Intake that were not recorded in the EMR? @fiona855

Reply:
Oh, now I realized that you missed our discussion from yesterday, @mogoodrich ?

Reply:
Oh, was there a discussion at standup that I missed? No worries, I don't mean to relitigate if you all have already discussed it.